src/json.rs

```diff
@@ -78,19 +92,31 @@ impl JsonValue {
     #[allow(dead_code)]
     pub fn parse_u8(v: &JsonValue) -> Result<u8, JsonError> {
         match v {
-            JsonValue::Number(n) => Ok(*n as u8),
+            JsonValue::Number(n) => {
+                if *n < u8::MIN as f64 || *n > u8::MAX as f64 {
+                    Err(JsonError::OutOfRangeU8)
+                } else {
+                    Ok(*n as u8)
+                }
+            }
             _ => Err(JsonError::InvalidU8),
         }
     }
 
     #[allow(dead_code)]
-    pub fn array<T, F>(&self, key: &str, call: F) -> Result<Vec<T>, JsonError>
+    pub fn array<T, F>(&self, key: &str, mut call: F) -> Result<Vec<T>, JsonError>
     where
-        F: FnMut(&JsonValue) -> T,
+        F: FnMut(&JsonValue) -> Result<T, JsonError>,
```

Comment:
It feels "off" to me to take an input as a Result here. I do not recommend it.

Reply:
Do you have another approach to catch errors from input functions with Result types?

Reply:
Well, do something with it before its passed to a method

Reply:
If this function were called `parse_array` I would be more on-board with this function signature. Parsing usually involves an error case.

I would expect a function simply called `array` to look something like `fn array(&self, key: &str) -> Result<Vec<JsonValue>, JsonError>`.

Reply:
Actually we don't use for release that function. So @birchmd I will try your solution.

Reply:
@birchmd I propose to delete that method. We do not use it.
Also implementation for `Vec<JsonValue>` has pitfalls.

Reply:
Works for me.